### ISSUE TITLE
Make configuration sections collapsible

### ISSUE THREAD

**Issue by pjdarton** [jenkinsci/docker-plugin#218]
I'd like the configuration page at http://my-jenkins/configure to have "collapsible" Docker sections.
i.e. make each "Docker" entry (within the "Cloud" section) default to be un-expanded, and make each "Images" entry default to be un-expanded.
i.e. the same way the "Advanced" button works on the "images" section.

At present, if one has multiple docker-host machines, each having multiple images, the http://my-jenkins/configure page becomes very long.  It becomes difficult to see the "big picture" because one cannot fit much more than one image definition on screen.
![existing](https://cloud.githubusercontent.com/assets/984537/7563175/7899c108-f7d2-11e4-945f-7898c53941e1.png)

If each section started out as "collapsed" then this web page would be much neater.
This technique has been used in other plugins, e.g. The "custom tool" plugin has a section "Custom tool" which is initially just a button "Custom tool installations...".  Similarly, JDK -> "JDK installations...", Ant -> "Ant installations" etc.

What I'd like is for the "Could" section to initially just contain a "Docker installations..." button,
![collapsed](https://cloud.githubusercontent.com/assets/984537/7563324/f927be78-f7d3-11e4-9141-14d14a7ba520.png)

each "Docker" section to be initially displayed as a button whose text was the "Name",
![desired_1_of_2](https://cloud.githubusercontent.com/assets/984537/7563182/81a6bd32-f7d2-11e4-938c-bdb380a6a1ef.png)

and each "Image" section to initially be displayed as a button whose text was the image's ID.
![desired_2_of_2](https://cloud.githubusercontent.com/assets/984537/7563187/85bc1994-f7d2-11e4-8b69-b0b0cc41c434.png)

i.e. make the configuration small by default, but expandable on demand.

**Comment by lanwen** [comment]
How it can be implemented in jelly?

**Comment by pjdarton** [comment]
I would hope and expect that it can be implemented in jelly in the same way that every other plugin (that implements this kind of functionality) implements it.
I would expect that, if one were to inspect the jelly code from those other plugins, this mechanism would become apparent.
(the last time I used jelly was on the accurev plugin, and it didn't seem very complicated there, so unless things have changed a lot, I don't expect this to be very complicated - I'd expect this to be a cosmetic change only, I'd expect the underlying data within the form to remain unchanged)

**Comment by lanwen** [comment]
How did you get this screenshots?

**Comment by KostyaSha** [comment]
@pjdarton feel free to contribute to core new forms or options that will allow docker plugin to use them.

**Comment by pjdarton** [comment]
The screenshots aren't "real" - I cheated - the screenshots were created using MSPaint - I figured that a picture would explain the sort of thing I wanted much better than English text (especially for people whose native language isn't English).

As for contributing, that's certainly an option, but would be an in-work thing hence subject to agreement of my boss(es) who might think I should concentrate my efforts elsewhere.  It's not out of the question (I've done so before on the accurev-jenkins plugin) but I can't make any promises on that front.  Right now, I've got sufficient "free time" to report issues & ask for enhancements, but not enough to do anything about those myself.  Hopefully that'll change and I'll be able to make some real contributions, as I believe that Jenkins + Docker is a very good thing to have...
